File: codewars/8kyu/regexp-basics-is-it-a-digit/dduck-regexp-basics-is-it-a-digit.js
// https://www.codewars.com/kata/regexp-basics-is-it-a-digit/

String.prototype.digit = function () {
  return !!this.match(/^[0-9]$/g);
};

// for multiple digit
// return (this.match(/[0-9]/g) && this.match(/[0-9]/g).length) === this.length;

// ''.digit();
//?
// , false);
'7'.digit();
//?
// , true);
// ' '.digit();
//?
//, false);
// 'a'.digit();
//?
//, false);
// 'a5'.digit();
//?
// //, false);
// '5a'.digit();
//?
// false

'14'.digit();
//?
//, false);

// describe('hex number', () => {
//   it('Basic tests', () => {
//     Test.assertEquals(''.digit(), false);
//     Test.assertEquals('7'.digit(), true);
//     Test.assertEquals(' '.digit(), false);
//     Test.assertEquals('a'.digit(), false);
//     Test.assertEquals('a5'.digit(), false);
//     Test.assertEquals('14'.digit(), false);
//   });
// });

File: codewars/8kyu/regexp-basics-is-it-a-digit/dduck-regexp-basics-is-it-a-digit.py
# https://www.codewars.com/kata/regexp-basics-is-it-a-digit/

import re


def is_digit(n):
    return re.search("^\d$", n) != None and len(n) == 1


print(is_digit(""), False)
print(is_digit("7"), True)
print(is_digit(" "), False)
print(is_digit("a"), False)
print(is_digit("a5"), False)
print(is_digit("1"), True)
print(is_digit(" 1"), True)
print(is_digit("1 "), True)
